### DISCUSSION TITLE
Header versioning

### DISCUSSION THREAD

**Discussion by ghost** [tokio-rs/axum#2444] [Q&A] [answered]
### Summary

hi...
how can i implement header versioning and when if not set api version use the default last version exist . 
please guide me 

### axum version

0.7

**Comment by davidpdrsn** [comment, marked as answer]
You’ll wanna implement [`FromRequestParts`](https://docs.rs/axum/0.7.2/axum/extract/trait.FromRequestParts.html)

**Reply by davidpdrsn** [reply]
Have a look through the examples in the repo.
